### PR TITLE
feat(screener): use chainalysis

### PR DESCRIPTION
**Pull request by golangisfun123**
**Description**
A clear and concise description of the features you're adding in this pull request.

**Additional context**
Add any other context about the problem you're solving.

**Metadata**
- Fixes #[Link to Issue]

## Summary by CodeRabbit

- **New Features**
    - Integrated Chainalysis API for address risk quantification.
    - Added new endpoints to screen addresses without specifying rulesets.

- **Bug Fixes**
    - Updated address screening calls to remove deprecated parameters.

- **Refactor**
    - Removed TRM Labs dependency and related interfaces.
    - Simplified database interface by focusing on blacklisted addresses.

- **Documentation**
    - Updated documentation to reflect changes from TRM Labs to Chainalysis API.
    - Modified Swagger API documentation for new endpoint structures and descriptions.

- **Chores**
    - Updated dependencies in `go.mod`.
    - Added `.gitignore` entry to exclude `config.yml`.

### CODE REVIEW

File: contrib/screener-api/screener/screener.go [hunk 1]
```diff
@@ -65,43 +60,61 @@ func NewScreener(ctx context.Context, cfg config.Config, metricHandler metrics.H
...
 	}
-
 	screener.db, err = sql.Connect(ctx, dbType, cfg.Database.DSN, metricHandler)
 	if err != nil {
 		return nil, fmt.Errorf("could not connect to rules db: %w", err)
 	}
 
 	screener.router = ginhelper.New(logger)
 	screener.router.Use(screener.metrics.Gin())
-	screener.router.Handle(http.MethodGet, "/:ruleset/address/:address", screener.screenAddress)
```
Comment (trajan0x) [inline contrib/screener-api/screener/screener.go:95]: we need to have a /:ruleset/address/:address here where ruleset is a no-op for backwards compatibility

File: contrib/screener-api/screener/screener.go [hunk 2]
```diff
@@ -65,43 +60,61 @@ func NewScreener(ctx context.Context, cfg config.Config, metricHandler metrics.H
...
 	screener.router.GET("/swagger/*any", ginSwagger.WrapHandler(swaggerfiles.Handler))
 
+	gin.SetMode(gin.ReleaseMode)
+
 	return &screener, nil
 }
 
+func (s *screenerImpl) Start(ctx context.Context) error {
+	// TODO: potential race condition here, if the blacklist is not fetched before the first request
+	// in practice trm will catch
```
Comment (trajan0x) [inline contrib/screener-api/screener/screener.go:94]: trm->chainalyiss

File: contrib/screener-api/screener/screener.go [hunk 3]
```diff
@@ -65,43 +60,61 @@ func NewScreener(ctx context.Context, cfg config.Config, metricHandler metrics.H
...
+	go func() {
+		for {
+			if s.cfg.BlacklistURL != "" {
+				s.fetchBlacklist(ctx)
+				time.Sleep(time.Second * 15)
```
Comment (trajan0x) [inline contrib/screener-api/screener/screener.go:99]: should respect context, e.g.:
```go
waitTime := time.Duration(0)
for{
select {

case <-ctx.Done():
return
case <-time.After(waitTime):
        waitTime = time.Second * 15
	s.fetchBlacklist(ctx)
}
```

File: contrib/screener-api/screener/screener.go [hunk 4]
```diff
@@ -65,43 +60,61 @@ func NewScreener(ctx context.Context, cfg config.Config, metricHandler metrics.H
...
+			}
+		}
+	}()
+	connection := baseServer.Server{}
+	err := connection.ListenAndServe(ctx, fmt.Sprintf(":%d", s.cfg.Port), s.router)
+	if err != nil {
+		return fmt.Errorf("could not start server: %w", err)
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (trajan0x) [reply contrib/screener-api/screener/screener.go:108]: don't want server to crash if can't be fetched

File: contrib/screener-api/screener/screener.go [hunk 5]
```diff
@@ -124,14 +136,63 @@ func (s *screenerImpl) fetchBlacklist(ctx context.Context) {
...
 	}
 
-	s.blacklistMux.Lock()
-	defer s.blacklistMux.Unlock()
-
+	s.blacklistCacheMux.Lock()
+	defer s.blacklistCacheMux.Unlock()
 	for _, item := range blacklist {
-		s.blacklist = append(s.blacklist, strings.ToLower(item))
+		s.blacklistCache[item] = true
```
Comment (trajan0x) [inline contrib/screener-api/screener/screener.go:142]: strings.ToLower(item)

**Review by trajan0x — CHANGES_REQUESTED** [review]

File: contrib/screener-api/screener/screener.go (hunk 1 above)
Comment (trajan0x) [reply contrib/screener-api/screener/screener.go:95]: in practice this is just:

```go
   // TODO: deprecated and ruelset is a no-op, this is for backwards compatibility and may be removed in a future version.
    screener.router.Handle(http.MethodGet, "/:ruleset/address/:address", screener.screenAddress)
```

File: contrib/screener-api/screener/screener.go (hunk 3 above)
Comment (golangisfun123) [reply contrib/screener-api/screener/screener.go:99]:
```
	go func() {
		// fetch the blacklist at start
		s.fetchBlacklist(ctx)

		for {
			select {
			case <-ctx.Done():
				return
			case <-time.After(time.Second * 15):
				s.fetchBlacklist(ctx)
			}
		}
	}()
```

File: contrib/screener-api/chainalysis/chainalysisapi.go [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+	}
+
+	// If the user is not registered, register them and try again.
+	if userNotRegistered(rawResponse) {
+		if err = c.registerAddress(ctx, address); err != nil {
+			return false, fmt.Errorf("could not register address: %w", err)
+		}
+
+		// Then try again.
+		time.Sleep(2 * time.Second)
```
Comment (trajan0x) [inline contrib/screener-api/chainalysis/chainalysisapi.go:80]: @golangisfun123 what's up with the wait here? This should be atomic

**Review by trajan0x — CHANGES_REQUESTED** [review]

File: contrib/screener-api/chainalysis/chainalysisapi.go (hunk 6 above)
Comment (golangisfun123) [reply contrib/screener-api/chainalysis/chainalysisapi.go:80]: i thought for latency so we don't fetch before it's registered

**Comment by trajan0x** [comment]
staged to fe in #2780

**Pull request merged**